### PR TITLE
Fix intercom issues

### PR DESCRIPTION
**Pull request by TheMagnetar**
**When merged this pull request will:**
- Fix: Intercom disconnect message is shown always when exiting a vehicle. 
- Better function naming.
- Fix "inside" keyword.
- Moved common functions in sys_rack and sys_intercom to sys_core

### CODE REVIEW

**Comment by TheMagnetar** [comment]
Ready

File: addons/sys_core/fnc_hasAccessToVehicleSystem.sqf [hunk 1]
```diff
@@ -0,0 +1,114 @@
+/*
+ * Author: ACRE2Team
+ * Checks if the given unit can have acess to a vehicle system like intercom or vehicle racks.
```
Comment (Soldia1138) [inline addons/sys_core/fnc_hasAccessToVehicleSystem.sqf]: `acess` -> `access`

File: addons/sys_core/fnc_processVehicleSystemAccessArray.sqf [hunk 2]
```diff
@@ -10,7 +35,7 @@
  * Formatted array <ARRAY>
  *
  * Example:
- * [cursorTarget, ["driver", ["cargo", 1, 2]]] call acre_sys_intercom_processConfigArray
+ * [cursorTarget, ["driver", ["cargo", 1, 2]]] call acre_sys_intercom_processVehicleSystemAccessArray
```
Comment (Soldia1138) [inline addons/sys_core/fnc_processVehicleSystemAccessArray.sqf]: `acre_sys_intercom_processVehicleSystemAccessArray` -> `acre_sys_intercom_fnc_processVehicleSystemAccessArray`

File: addons/sys_intercom/fnc_enterVehicle.sqf [hunk 3]
```diff
@@ -61,5 +61,7 @@ if (_unit != _vehicle) then {
     _unit setVariable [QGVAR(intercomVehicle), objNull];
     ACRE_PLAYER_INTERCOM = [];
 
-    ["Disconnected from intercom system", ICON_RADIO_CALL] call EFUNC(sys_core,displayNotification);
+    if (_disconnected) then {
+        ["Disconnected from intercom system", ICON_RADIO_CALL] call EFUNC(sys_core,displayNotification);
```
Comment (jonpas) [inline addons/sys_intercom/fnc_enterVehicle.sqf]: Stringtable?

File: addons/sys_core/fnc_processVehicleSystemAccessArray.sqf [hunk 4]
```diff
@@ -1,6 +1,31 @@
...
+ *   vehicle role.
+ * - cargo: identifier for cargo positions. It must be followed by a valid cargo index or the "all" keyword.
+ * - turret: identifier for turret positions. It must be followed by a valid turret path or the "all" keyword.
+ * - ffv: identifier for turret positions of the type ffv. It must be followed by a valid turret path or the "all" keyword.
+ * - turnedOut: identifier for situations where the player is turned out of the vehicle. It must be followed by a valid
+ *   turret path or the "all" keyword.
+ * - inside: selects all seats inside the vehicle.
+ *
+ * The following cases are ilustrative examples of the outcome of this function:
+ *
```
Comment (jonpas) [inline addons/sys_core/fnc_processVehicleSystemAccessArray.sqf]: Same as above.

File: addons/sys_core/fnc_processVehicleSystemAccessArray.sqf [hunk 5]
```diff
@@ -1,6 +1,31 @@
...
  * Author: ACRE2Team
- * Parses the intercom configuration arrays and converts it to a usable format.
+ * Parses the intercom and racks configuration arrays and converts it to a usable format.
+ *
+ * The configuration arrays are used in order to determine seat functionality in a vehicle using high level entries like
+ * "all", "inside", "crew". This function converts these configuration arrays to a format that is more easy to parse in
+ * code.
+ *
+ * The following keywords are supported:
+ *
```
Comment (jonpas) [inline addons/sys_core/fnc_processVehicleSystemAccessArray.sqf]: No need for this empty line, unnecessarily splits it.

File: addons/sys_core/XEH_PREP.hpp [hunk 6]
```diff
@@ -51,4 +52,3 @@ PREP(enableRevealAI);
 PREP(onRevealUnit);
 
 PREP(coreInitPFH);
-//PREP(vehicleCrewPFH);
```
Comment (jonpas) [inline addons/sys_core/XEH_PREP.hpp:54]: ?

File: addons/sys_core/fnc_hasAccessToVehicleSystem.sqf [hunk 7]
```diff
@@ -0,0 +1,114 @@
+/*
+ * Author: ACRE2Team
+ * Checks if the given unit can have access to a vehicle system like intercom or vehicle racks.
+ *
+ * Arguments:
+ * 0: Vehicle <OBJECT>
+ * 1: Unit <OBJECT>
+ * 2: Allowed positions <ARRAY>
+ * 3: Forbidden positions <ARRAY>
+ * 4: Maximum distance for external use <NUMBER><OPTIONAL>
```
Comment (jonpas) [inline addons/sys_core/fnc_hasAccessToVehicleSystem.sqf]: ` * 4: Maximum distance for external use <NUMBER> (default: 0)`

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/sys_core/XEH_PREP.hpp (hunk 6 above)
Comment (TheMagnetar) [reply addons/sys_core/XEH_PREP.hpp:54]: Remnant of an old PFH... it was deprecated with the first push of the new intercom + rack system

**Review by jonpas — APPROVED** [review]

**Pull request merged**